### ISSUE TITLE
systemd-networkd: RFC4191 not being honored for 2 received RAs with different preferences

### ISSUE THREAD

**Issue by mwetterw** [systemd/systemd#25138]
### systemd version the issue has been seen with

251.7-1-arch

### Used distribution

Arch Linux

### Linux kernel version used

6.0.2-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

Two default routes with a clear priority preference difference between them, making the highest preference route being always used, and the lower preference route being never used.

### Unexpected behaviour you saw

Both routes appear but are equal cost. This results in ECMP behavior.

### Steps to reproduce the problem

Have two IPv6 routers in the scope link, each of them sending RAs.
- One of the router sending RA with preference medium
- The other router sending RA with preference high

### Additional program output to the terminal or log subsystem illustrating the issue

(IPs have been modified)
```sh
ip -c -6 -br r
::1 dev lo proto kernel metric 256 pref medium
2a00:abc:def:beef::/64 dev wlo1 proto ra metric 1024 expires 2591396sec pref medium
fe80::/64 dev wlo1 proto kernel metric 256 pref medium
default proto ra metric 1024 expires 1596sec mtu 1500 pref medium
	nexthop via fe80::2021:22ff:fe23:2425 dev wlo1 weight 1 
	nexthop via fe80::2526:27ff:fe28:2930 dev wlo1 weight 1
```
```sh
cat /etc/systemd/network/25-wireless.network 
[Match]
Name=wlo1

[Link]
MTUBytes=1350

[Network]
DHCP=yes
```

**Comment by yuwata** [comment]
Fix is waiting in #25166.

**Comment by mwetterw** [comment]
Nice! Thank you!

Do you know if after this fix, we'll conform like host type C of the RFC?
The bug I noticed was only about preferences of the whole RA message (in the header). But does systemd-networkd have the correct behavior also about preferences of the Routes themselves embedded inside the RAs?

If I understand correctly, if 2 routers announce exactly the same route, in that case we don't care about the global RA preference of the router, but care about the embedded route preference to decide which router wins on a specific route.

**Comment by yuwata** [comment]
@mwetterw 
> Do you know if after this fix, we'll conform like host type C of the RFC?

We do not use Type A, B, C. All information is always examined. Users can control which information is used by e.g. `UseGateway=`, `UseOnlinePrefix=` or so.

> The bug I noticed was only about preferences of the whole RA message (in the header). But does systemd-networkd have the correct behavior also about preferences of the Routes themselves embedded inside the RAs?
>
> If I understand correctly, if 2 routers announce exactly the same route, in that case we don't care about the global RA preference of the router, but care about the embedded route preference to decide which router wins on a specific route.

That should be also addressed by #25166. The fix is applied to both the default route (built from header) and routes built from each route prefix.
